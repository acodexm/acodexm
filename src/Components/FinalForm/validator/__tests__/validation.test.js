import {
  isRequired,
  maxLength,
  maxValue,
  minLength,
  minValue,
  mustBeEmail,
  mustBeNumber,
} from '../validator';

describe('check custom validators', () => {

  it('should validate isRequired', () => {
    expect(isRequired(false)()).toMatchSnapshot();
    expect(isRequired(false)('')).toMatchSnapshot();
    expect(isRequired(false)('test')).toMatchSnapshot();
    expect(isRequired(true)('test')).toMatchSnapshot();
    expect(isRequired()('test')).toMatchSnapshot();
    expect(isRequired()(true)).toMatchSnapshot();
    // incorrect
    expect(isRequired(true)()).toMatchSnapshot();
    expect(isRequired(true)('')).toMatchSnapshot();
    expect(isRequired()()).toMatchSnapshot();
    expect(isRequired()('')).toMatchSnapshot();
    expect(isRequired()(null)).toMatchSnapshot();
    expect(isRequired()(undefined)).toMatchSnapshot();
    expect(isRequired()(0)).toMatchSnapshot();
    expect(isRequired()(false)).toMatchSnapshot();
  });
  it('should validate mustBeNumber', () => {
    expect(mustBeNumber(1)).toMatchSnapshot();
    expect(mustBeNumber('NaN')).toMatchSnapshot();
  });
  it('should validate minValue', () => {
    expect(minValue(10)(10)).toMatchSnapshot();
    expect(minValue(10)(1)).toMatchSnapshot();
    expect(minValue(10)()).toMatchSnapshot();
  });
  it('should validate maxValue', () => {
    expect(maxValue(10)(10)).toMatchSnapshot();
    expect(maxValue(10)(100)).toMatchSnapshot();
    expect(maxValue(10)()).toMatchSnapshot();
  });
  it('should validate minLength', () => {
    expect(minLength(10)('ten chars!')).toMatchSnapshot();
    expect(minLength(10)('less 10')).toMatchSnapshot();
    expect(minLength(10)()).toMatchSnapshot();
  });
  it('should validate maxLength', () => {
    expect(maxLength(10)('ten chars!')).toMatchSnapshot();
    expect(maxLength(10)('over ten chars!')).toMatchSnapshot();
    expect(maxLength(10)()).toMatchSnapshot();
  });
  it('should validate mustBeEmail', () => {
    expect(mustBeEmail('a@a.a')).toMatchSnapshot();
    expect(mustBeEmail('aa.a')).toMatchSnapshot();
    expect(mustBeEmail()).toMatchSnapshot();
  });
});
