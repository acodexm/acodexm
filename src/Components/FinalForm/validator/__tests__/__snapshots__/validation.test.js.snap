// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check custom validators should validate isRequired 1`] = `undefined`;

exports[`check custom validators should validate isRequired 2`] = `undefined`;

exports[`check custom validators should validate isRequired 3`] = `undefined`;

exports[`check custom validators should validate isRequired 4`] = `undefined`;

exports[`check custom validators should validate isRequired 5`] = `undefined`;

exports[`check custom validators should validate isRequired 6`] = `undefined`;

exports[`check custom validators should validate isRequired 7`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 8`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 9`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 10`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 11`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 12`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 13`] = `"This field is required"`;

exports[`check custom validators should validate isRequired 14`] = `"This field is required"`;

exports[`check custom validators should validate maxLength 1`] = `undefined`;

exports[`check custom validators should validate maxLength 2`] = `"validation.max"`;

exports[`check custom validators should validate maxLength 3`] = `undefined`;

exports[`check custom validators should validate maxValue 1`] = `undefined`;

exports[`check custom validators should validate maxValue 2`] = `"validation.max.num"`;

exports[`check custom validators should validate maxValue 3`] = `undefined`;

exports[`check custom validators should validate minLength 1`] = `undefined`;

exports[`check custom validators should validate minLength 2`] = `"validation.min"`;

exports[`check custom validators should validate minLength 3`] = `undefined`;

exports[`check custom validators should validate minValue 1`] = `undefined`;

exports[`check custom validators should validate minValue 2`] = `"validation.min.num"`;

exports[`check custom validators should validate minValue 3`] = `undefined`;

exports[`check custom validators should validate mustBeEmail 1`] = `"Wrong email address"`;

exports[`check custom validators should validate mustBeEmail 2`] = `"Wrong email address"`;

exports[`check custom validators should validate mustBeEmail 3`] = `undefined`;

exports[`check custom validators should validate mustBeNumber 1`] = `undefined`;

exports[`check custom validators should validate mustBeNumber 2`] = `"validation.number"`;
